service/project.go

```diff
@@ -341,7 +341,17 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	//modify project vars if necessary
+	// If the variable is private , and if the variable in the submission is
+	// empty, then do not modify it. This variable has been redacted and is
+	// therefore empty in the submission, since the client does not have
+	// access to it, and we should not overwrite it.
+	for k, v := range projectVars.Vars {
+		if _, ok := projectVars.PrivateVars[k]; ok {
+			if val, ok := responseRef.ProjVarsMap[k]; val == "" && ok {
+				responseRef.ProjVarsMap[k] = v
```

Comment:
So one thing here: this is means I can't clear the value of a private var right?

Reply:
Yes, you can, since line 350 only sets the variable if it's in the submission's project vars map, which it will not be if it has been deleted.

Reply:
(I tested a bunch of permutations of this locally, and they were all happy.)